### PR TITLE
Initial connector component with GQL schema

### PR DESCRIPTION
**Pull request by crabtree**
**Description**

Changes proposed in this pull request:

- The initial draft for connector GraphQL schema
- Connector component structure

**Related issue(s)**

### CODE REVIEW

File: components/connector/internal/gqlschema/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Certificate
```
Comment (PK85) [inline components/connector/internal/gqlschema/schema.graphql]: Could you already think and propose a concept of generic connector APIs. I mean for example JWT too.

Comment (crabtree) [reply components/connector/internal/gqlschema/schema.graphql]: Sure, will do.

File: components/connector/internal/gqlschema/schema.graphql [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+type Query {
+    # Client-Certificates
+    
+    # Retuns subject that should be placed in the signing request
+    getCSRInfo(token: String): CSRInfo!
+}
+
+type Mutation {
+    # Applications
+    getApplicationToken(appID: ID!): Token! # internal?
```
Comment (pkosiec) [inline components/connector/internal/gqlschema/schema.graphql]: hmm, maybe `generate` would be a better name. At first I thought that this is simple query, but of course it generates the token under the hood.

File: components/connector/internal/gqlschema/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+# CSR
+type CSR {
+    subject: String! # eg.: "OU=Test,O=Test,L=Blacksburg,ST=Virginia,C=US,CN={ID}"
+    keyAlgorithm: String! # eg.: rsa2048
+}
+
+# CSRInfo
+type CSRInfo {
+    csrURL: String! # eg.: "https://connector-service.test.cluster.kyma.cx/v1/runtimes/certificates?token=1edfc34g"
```
Comment (pkosiec) [inline components/connector/internal/gqlschema/schema.graphql]: hmm, maybe we should put this URL into `CSR` type.

Comment (pkosiec) [reply components/connector/internal/gqlschema/schema.graphql]: Also I would rather not use abbreviations for type names. I would go with the `CertificateSigningRequest` name

File: components/connector/internal/gqlschema/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+# Certificate
+type Certificate {
+    crt: String!
+    caCrt: String!
+    clientCrt: String!
+}
+
+# MPInfo
+type MPInfo {
```
Comment (pkosiec) [inline components/connector/internal/gqlschema/schema.graphql]: hmm, what about `ManagementPlaneConfiguration`, `ManagementPlaneDetails`, or something like that?

File: components/connector/internal/gqlschema/schema.graphql [hunk 5]
```diff
@@ -0,0 +1,59 @@
+# Token
+type Token {
+    url: String! # eg.: "https://connector-service.test.cluster.kyma.cx/v1/applications/info?token=1edfc34g"
+    token: String! # eg.: "1edfc34g"
+}
+
+# Certificate
+type Certificate {
```
Comment (pkosiec) [inline components/connector/internal/gqlschema/schema.graphql]: I would go with full names:

```graphql
certificate: String!
certificateAuthority: String!
clientCertificate: String!
```

File: components/connector/internal/gqlschema/schema.graphql (hunk 3 above)
Comment (crabtree) [reply components/connector/internal/gqlschema/schema.graphql]: After further considerations, this URL is not relevant here, so I remove it.

File: components/connector/internal/gqlschema/schema.graphql (hunk 5 above)
Comment (crabtree) [reply components/connector/internal/gqlschema/schema.graphql]: I changed it slightly, please take a look

File: components/connector/internal/gqlschema/schema.graphql (hunk 4 above)
Comment (crabtree) [reply components/connector/internal/gqlschema/schema.graphql]: I changed it to ManagementPlaneInfo

File: components/connector/internal/gqlschema/schema.graphql [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+type CertificateSigningRequestInfo {
+    managementPlaneInfo: ManagementPlaneInfo!
+    subject: String! # eg.: "OU=Test,O=Test,L=Blacksburg,ST=Virginia,C=US,CN={ID}"
+    keyAlgorithm: String! # eg.: rsa2048
+}
+
+type Query {
+    # Client-Certificates
+    
+    # Retuns subject that should be placed in the signing request
```
Comment (pkosiec) [inline components/connector/internal/gqlschema/schema.graphql]: Returns

Comment (pkosiec) [reply components/connector/internal/gqlschema/schema.graphql]: Please write comments in `"""comment"""` form, to include it in generated API documentation

File: components/connector/internal/gqlschema/schema.graphql [hunk 7]
```diff
@@ -0,0 +1,50 @@
+# Token
+type Token {
+    token: String! # eg.: "1edfc34g"
+}
+
+# Certificates
+type Certificates {
```
Comment (pkosiec) [inline components/connector/internal/gqlschema/schema.graphql]: I would think about better name for the type (I don't like the plural form), but `CertificationDetails` is all I can made up for now 😄 and I don't think it's is better than yours, so...

Comment (crabtree) [reply components/connector/internal/gqlschema/schema.graphql]: Renamed to `CertificationResult`

**Pull request merged**